### PR TITLE
Allow for filtering of administrative tasks

### LINKED ISSUE #1130: Do not forcefully include admin tasks when using --include-tasks and --exclude tasks
#1116 fixed a broken behavior where specifying `--exclude-tasks` also resulted excluding [administrative tasks](https://github.com/elastic/rally/blob/72fbb94a803ea476048290d31d0d5d742e6b7451/esrally/track/track.py#L528-L561).

While there are historical reasons behind this (primarily, implicitly creating indices for indices defined in the [index section](https://github.com/elastic/rally/blob/72fbb94a803ea476048290d31d0d5d742e6b7451/esrally/track/track.py#L528-L561)) this is no problem anymore, we need to specify explicit operations to create indices [already from version 0.9.0](https://esrally.readthedocs.io/en/stable/migrate.html#migrating-to-rally-0-9-0).

We should allow full user control over which operations should be executed when using `--include-tasks` or `--exclude-tasks`.

## Implementation hints

1. Forcefully including admin tasks is done [here](https://github.com/elastic/rally/blob/72fbb94a803ea476048290d31d0d5d742e6b7451/esrally/track/loader.py#L713-L714)
2. Administrative actions are specified [here](https://github.com/elastic/rally/blob/72fbb94a803ea476048290d31d0d5d742e6b7451/esrally/track/track.py#L528-L561)
3. Unit tests for task filtering can be found [here](https://github.com/elastic/rally/blob/72fbb94a803ea476048290d31d0d5d742e6b7451/tests/track/loader_test.py#L1293-L1488)
4. There is also an IT test [here](https://github.com/elastic/rally/blob/72fbb94a803ea476048290d31d0d5d742e6b7451/it/info_test.py#L32-L34). IT tests are "expensive" (i.e. take time to stand up/tear things down) so adding additional IT tests should be considered carefully and prefer, if possible, unit tests.
5. Docs for the two filtering cli arguments can be found [here](https://github.com/elastic/rally/blame/master/docs/command_line_reference.rst#L324-L352) and should be updated accordingly. Changes in behavior should also get documented [here](https://github.com/elastic/rally/blob/master/docs/migrate.rst).
6. Manual test is quite easy, without the need to actually run a benchmark (even without `--test-mode`) by just leveraging the [info subcommand](https://esrally.readthedocs.io/en/stable/command_line_reference.html#info) as shown in [this IT test](https://github.com/elastic/rally/blob/72fbb94a803ea476048290d31d0d5d742e6b7451/it/info_test.py#L32-L34).

Relates #1116 / https://github.com/elastic/rally/commit/f06991747ae48064d1433fd72d3b6ee3f4ceeb08

### PR DESCRIPTION
**Pull request by DJRickyB**
* modify tests for new behavior, add tests to preserve behavior
* delete AdminTaskFilter and its usages
* update documentation

Resolves #1130

### CODE REVIEW

File: docs/command_line_reference.rst [hunk 1]
```diff
@@ -326,7 +326,7 @@ A comma-separated list if IP:transport port pairs used to specify the seed hosts
 
 Each challenge consists of one or more tasks but sometimes you are only interested to run a subset of all tasks. For example, you might have prepared an index already and want only to repeatedly run search benchmarks. Or you want to run only the indexing task but nothing else.
 
-You can use ``--include-tasks`` to specify a comma-separated list of tasks that you want to run. Each item in the list defines either the name of a task or the operation type of a task. Only the tasks that match will be executed. Currently there is also no command that lists the tasks of a challenge so you need to look at the track source.
+You can use ``--include-tasks`` to specify a comma-separated list of tasks that you want to run. Each item in the list defines either the name of a task or the operation type of a task. Only the tasks that match will be executed. You can use the ``info`` subcommand to list the tasks of a challenge, or look at the track source.
```
Comment (danielmitterdorfer) [inline docs/command_line_reference.rst:329]: Great you spotted this!

File: docs/migrate.rst [hunk 2]
```diff
@@ -58,6 +58,11 @@ bulk-size metrics property is dropped
 
 Metrics records for bulk request don't contain the ``bulk-size`` property anymore. Please use the ``weight`` property instead and consider the ``unit`` property to interpret the value.
 
+--include-filters and --exclude-filters affect all operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Prior to 2.0.3, administrative tasks/operations (see :ref:`operations documentation<track_operations>`) were exempt from filtering and would run regardless of filtering.  ``--include-filters`` and ``--exclude-filters`` flags now can affect all operations in a track.  If you make use of include filters, it is advised to check that all desired operations are listed.
```
Comment (danielmitterdorfer) [inline docs/migrate.rst:64]: The respective command line flags are not called `--include-filters` / `--exclude-filters` but rather `--include-tasks` and `--exclude-tasks`. Also, as the filters are referring to tasks, I'd start this sentence with "Prior to 2.0.3, administrative tasks (see ...)".

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks for fixing this! I spotted a small issue in the migration docs but the actual change looks fine.

File: docs/migrate.rst (hunk 2 above)
Comment (DJRickyB) [reply docs/migrate.rst]: fixed

File: docs/migrate.rst [hunk 3]
```diff
@@ -61,7 +61,7 @@ Metrics records for bulk request don't contain the ``bulk-size`` property anymor
 --include-filters and --exclude-filters affect all operations
```
Comment (danielmitterdorfer) [inline docs/migrate.rst:61]: I'm sorry I did not point it out initially but my comment about `--include-tasks` and `--exclude-tasks` also applies to the heading here.

**Review by danielmitterdorfer — COMMENTED** [review]
One more change and we're good. :)

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for iterating so quickly. LGTM

File: docs/migrate.rst [hunk 4]
```diff
@@ -58,6 +58,11 @@ bulk-size metrics property is dropped
 
 Metrics records for bulk request don't contain the ``bulk-size`` property anymore. Please use the ``weight`` property instead and consider the ``unit`` property to interpret the value.
 
+--include-tasks and --exclude-tasks affect all operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Prior to 2.0.3, administrative tasks (see :ref:`operations documentation<track_operations>`) were exempt from filtering and would run regardless of filtering.  ``--include-tasks`` and ``--exclude-tasks`` flags now can affect all operations in a track.  If you make use of include filters, it is advised to check that all desired operations are listed.
```
Comment (dliappis) [inline docs/migrate.rst:64]: nit: We use one space after a period (consistently), so let's stick to one here too.

**Review by dliappis — APPROVED** [review]
LGTM thank you! Left one nit.

File: docs/migrate.rst (hunk 4 above)
Comment (DJRickyB) [reply docs/migrate.rst]: fixed! and unlike last time I searched for all occurrences and fixed both

**Pull request merged**